odoo/fields.py

```diff
@@ -249,6 +249,8 @@ def __init__(self, string=Default, **kwargs):
         kwargs['string'] = string
         self._sequence = kwargs['_sequence'] = next(_global_seq)
         self.args = {key: val for key, val in kwargs.items() if val is not Default}
+        self.name = type(self).__name__
+        self.model_name = '_unknown'
```

Comment:
`str(field)` no longer crashes, instead it returns `"None.None"`.
To me `"_unknown.Integer"` is not clearer.  Why not special-casing `__str__` to return `"<fields.Integer>"` instead?

Reply:
Now it shows `<odoo.fields.Char>`